tests/dragonfly/__init__.py

```diff
@@ -28,3 +28,15 @@ def is_port_available(self, port):
 
         with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
             return s.connect_ex(("localhost", port)) != 0
+
+
+@pytest.fixture(autouse=True)
+def run_before_and_after_test():
+    # Setup: logic before any of the test starts
+    # Empty the log on each run
+    last_log_file = open("/tmp/last_test_log_files.txt", "w")
+    last_log_file.close()
+
+    yield  # this is where the testing happens
+
```

Comment:
if you place this in conftest, you don't need to import it everywhere

Reply:
I would take an oath that I tried this and it did not work and I ended up with the changes here but I retried now and it did :facepalm: :facepalm: :face_exhaling: